src/compiler/sys/config.ts

```diff
@@ -4,11 +4,10 @@ import { createSystem } from './stencil-sys';
 import { setPlatformPath } from '../sys/modules/path';
 
 export const getConfig = (userConfig: d.Config): d.ValidatedConfig => {
```

Comment:
> It seems to me that it's used there as a consequence of the userConfig argument being of type Config there, but if that argument were to be changed to ValidatedConfig then I think that we could potentially delete this function? 

Exactly - because folks can call this from JS, and we haven't validated the config yet, we use continue to use `getConfig` to ensure fields that are needed are backfilled if the user doesn't provide them. The primary difference between `getConfig` and `validateConfig` that concerns me a little right now is that `getConfig` backfills only a subset of fields that  `validateConfig` does.  To date, we've been "lucky" that the fields we've made required were, well "required" to run Stencil. I hope that we don't run into a situation where `getConfig` and `validateConfig` diverge what they return as we make more and more fields required on `ValidatedConfig` (but it's always a possibility). As long as they don't, we should be well positioned to delete `getConfig` once we've made the remaining decisions regarding required field on `ValidatedConfig`.

At least that's what I'd been thinking to date - continue making `ValidatedConfig` stricter and once we had accomplished that, consider removing `getConfig`

Reply:
that makes sense to me. I think in general we want to live in a world where there is a single gate that a configuration must pass through to get from whatever the user is supplying to the actual compiler internals — `validateConfig` will be that gate!

ideally we could model the program something like

```sh
cat stencil.config.ts | validateConfig | theRestOfTheProgram
```

it sounds like we're on the same page about that, but that we don't want to get rid of `getConfig` quite yet, which makes sense to me 👍

Reply:
Yeah, absolutely 💯 💯